source/MaterialXGenOgsXml/OgsFragment.cpp

```diff
@@ -54,7 +54,7 @@ class GlslGeneratorWrapperBase
 
         // Maya images require a texture coordinates to be flipped in V.
         genOptions.fileTextureVerticalFlip = true;
-        genOptions.hwTransparency = mx::isTransparentSurface(_element, generator);
+        genOptions.hwTransparency = _isSurface && mx::isTransparentSurface(_element, generator);
```

Comment:
Somehow, `mx::isTransparentSurface()` returns that true for some non-surfaces, e.g. `rgb_to_hsv_to_rgb_color4_out`. This produces GLSL parsing errors during HLSL cross-compilation due to trying to return RGB as an RGBA from the root function:
`ERROR: 1:198: 'return' : cannot convert return value to function return type 
WARNING: 1:198: 'return' : type conversion on return values was not explicitly allowed until version 420 
ERROR: 2:0: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.`

Reply:
I looked at the produced code and it seems to be return 4 channel RGBA properly. I guess you have something in your converter which is producing 3 channel for some reason?
```
void main()
{
    vec4 rgbtohsv1_out = vec4(0.0);
    mx_rgbtohsv_color4(rgbtohsv1_in, rgbtohsv1_out);
    vec4 hsvtorgb1_out = vec4(0.0);
    mx_hsvtorgb_color4(rgbtohsv1_out, hsvtorgb1_out);
    out1 = hsvtorgb1_out;
}
```
out1 is declared as:
```
out vec4 out1;
```
BTW the name of the method isn't the greatest but 4 channel output is consider to possibly be transparent. It does not need to be a surface shader.

Reply:
Here's the GLSL output of the fragment generator:
`vec4 rgb_to_hsv_to_rgb_color4_out
(
    vec4 rgbtohsv1_in,
    float vp2Transparency
)
{
    vec4 rgbtohsv1_out = vec4(0.0);
    mx_rgbtohsv_color4(rgbtohsv1_in, rgbtohsv1_out);
    vec4 hsvtorgb1_out = vec4(0.0);
    mx_hsvtorgb_color4(rgbtohsv1_out, hsvtorgb1_out);
    return hsvtorgb1_out.xyz;
}`

So it must be a simple fix, if alpha is supposed to represent opacity.

Reply:
I think you're waiting for me, and I'm waiting for you. I guess get the fix for this in, and instead of waiting for me get it approved by Niklas :).

Reply:
I was just going to commit the proper fix with textured opacity after the summit but @JGamache-autodesk beat me to it :-) (#700). So I'll just revert this fix and merge the lighting-related fix because it seems uncontroversial.